pages/Article.jsx

```diff
@@ -0,0 +1,41 @@
+import { useParams } from 'react-router-dom'
+import ArticleCard from '../components/ArticleCard'
+import { useEffect, useState } from 'react'
+import { getArticle } from '../api'
+
+export default function Article () {
+    const [currentArticle, setCurrentArticle] = useState(null)
+    const [isError, setIsError] = useState(false)
+    const [isLoading, setIsLoading] = useState(true)
+
+    const  { article } = useParams()
+    useEffect(() => {
+        getArticle(article)
+        .then((res) => {
+            setCurrentArticle(res)
+            setIsError(false)
+        })
+        .catch(() => {
+            setIsError(true)
+        })
+        .finally(() => {
+            setIsLoading(false)
+        })
+    }, [])
```

Comment:
Not necessary to put the loading state in it's own block, it can be worked into the .then and catch